packages/kbn-url-state/use_sync_to_url.ts

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { useCallback, useEffect } from 'react';
+import { encode, decode } from '@kbn/rison';
+
+// https://developer.mozilla.org/en-US/docs/Web/API/Window/popstate_event
+const POPSTATE_EVENT = 'popstate' as const;
+
+/**
+ * Sync any object with browser query string using @knb/rison
+ *  @param key query string param to use
+ *  @param restore use this to handle restored state
+ *  @param cleanupOnHistoryNavigation use history events to cleanup state on back / forward naviation. true by default
+ */
+export const useSyncToUrl = <TValueToSerialize>(
```

Comment:
To the older point, can you make a reference to this in https://github.com/elastic/kibana/blob/main/x-pack/plugins/security_solution/public/common/hooks/use_url_state.ts ? The main thing for me is that all of the AET url search param logic is managed here currently so I don't want it to get lost. At some point we should unify all of this logic, but adding a reference for now so people are aware this is managed externally here would be great for documentation.

Reply:
:heavy_check_mark: